### PR TITLE
[MISC] deprecate seqan3::alignment_coordinate

### PR DESCRIPTION
**Pull request by SGSSGene**
Part of https://github.com/seqan/product_backlog/issues/282

### CODE REVIEW

**Comment by SGSSGene** [comment]
Open: many other functions use `seqan3::alignment_coordinate` and probably have to be freed from it?

**Review by MitraDarja — APPROVED** [review]
LGTM. But can you check, what the problem with the CI is?

File: test/unit/alignment/pairwise/fixture/alignment_fixture.hpp [hunk 1]
```diff
@@ -121,12 +124,15 @@ alignment_fixture(
...
-    seqan3::alignment_coordinate end_positions
+    size_t begin1_position,
+    size_t begin2_position,
+    size_t end1_position,
+    size_t end2_position
 )
 -> alignment_fixture<sequence1_t, sequence2_t, config_t, score_t,
                      std::vector<score_t>, std::vector<seqan3::detail::trace_directions>>;
 
+
```
Comment (Irallia) [inline test/unit/alignment/pairwise/fixture/alignment_fixture.hpp:135]: Multiple new lines

File: test/unit/alignment/pairwise/fixture/alignment_fixture.hpp [hunk 2]
```diff
@@ -62,8 +61,10 @@ struct alignment_fixture
     std::string aligned_sequence1;
     std::string aligned_sequence2;
 
-    seqan3::alignment_coordinate begin_positions;
-    seqan3::alignment_coordinate end_positions;
+    size_t begin1_position;
+    size_t begin2_position;
+    size_t end1_position;
+    size_t end2_position;
```
Comment (Irallia) [inline test/unit/alignment/pairwise/fixture/alignment_fixture.hpp:67]: Maybe give the variables clearer names:
```suggestion
    size_t begin_column_position;
    size_t begin_row_position;
    size_t end_column_position;
    size_t end_row_position;
```

File: test/unit/alignment/pairwise/edit_distance_unbanded_test_template.hpp [hunk 3]
```diff
@@ -164,7 +164,8 @@ TYPED_TEST_P(edit_distance_unbanded_test, end_positions)
     auto alignment = edit_distance<TypeParam::template edit_traits_type>(database, query, align_cfg);
     auto end_positions = alignment.end_positions();
 
-    EXPECT_EQ(end_positions, fixture.end_positions);
+    EXPECT_EQ(end_positions.first, fixture.end1_position);
+    EXPECT_EQ(end_positions.second, fixture.end2_position);
```
Comment (Irallia) [inline test/unit/alignment/pairwise/edit_distance_unbanded_test_template.hpp:168]:
```suggestion
    EXPECT_EQ(end_positions.first, fixture.end_column_position);
    EXPECT_EQ(end_positions.second, fixture.end_row_position);
```
or
```suggestion
    EXPECT_EQ(end_positions.first, fixture.end_first_position);
    EXPECT_EQ(end_positions.second, fixture.end_second_position);
```
?

File: test/unit/alignment/pairwise/fixture/local_affine_banded.hpp [hunk 4]
```diff
@@ -52,8 +52,11 @@ static auto dna4_01 = []()
         11,
         "GTTTA",
         "GTCTA",
-        seqan3::alignment_coordinate{column_index_type{5u}, row_index_type{2u}},
-        seqan3::alignment_coordinate{column_index_type{10u}, row_index_type{7u}},
+        // begin and end positions
```
Comment (Irallia) [inline test/unit/alignment/pairwise/fixture/local_affine_banded.hpp:55]: You could add this comment in the other tests aswell 
eg `global_affine_unbanded.hpp`

**Review by Irallia — CHANGES_REQUESTED** [review]
Looks very good already. I would only rename begin1, begin2, end1, end2.
Also, I like your comment, but you are just using it very randomly.
```cpp
 // begin and end positions
        0u,
        0u,
        22u,
        12u,
        ```

File: test/unit/alignment/pairwise/edit_distance_unbanded_test_template.hpp (hunk 3 above)
Comment (marehr) [reply test/unit/alignment/pairwise/edit_distance_unbanded_test_template.hpp:168]: They are called

* sequence1_begin_position
* sequence1_end_position
* sequence2_begin_position
* sequence2_end_position

https://docs.seqan.de/seqan/3-master-user/classseqan3_1_1alignment__result.html#a6f33588c817b822d0ebe7068d3072b6c

File: test/unit/alignment/pairwise/fixture/global_affine_banded.hpp [hunk 5]
```diff
@@ -55,8 +55,10 @@ static auto dna4_01 = []()
         -18,
         "A---ACCGGTTAACCGGTT",
         "ACGTAC----------GTA",
-        seqan3::alignment_coordinate{seqan3::detail::column_index_type{0u}, seqan3::detail::row_index_type{0u}},
-        seqan3::alignment_coordinate{seqan3::detail::column_index_type{16u}, seqan3::detail::row_index_type{9u}},
+        0u,
+        0u,
+        16u,
+        9u,
```
Comment (marehr) [inline test/unit/alignment/pairwise/fixture/global_affine_banded.hpp:61]: Maybe you can reapply your regex and add

```suggestion
        /*.sequence1_begin_position = */ 0u,
        /*.sequence2_begin_position = */ 0u,
        /*.sequence1_end_position = */ 16u,
        /*.sequence2_end_position = */ 9u,
```

**Comment by SGSSGene** [comment]
Very good suggestions! I adjusted my regex so it actually works correctly...changed it to `.sequence1_begin_position` also added the suggested comments.

**Review by Irallia — APPROVED** [review]
Perfect, thank you!

**Review by marehr — APPROVED** [review]

**Pull request merged**